### PR TITLE
Added PR Template

### PR DESCRIPTION
**Pull request by Warfields**
The template that's included here may be too prescriptive, I pulled it from an earlier project. Alternatively we could go for something like
```markdown
### Description

Please provide a brief description of the changes in this pull request.

### Stability
Keep in mind, There are rather large users of lol-html so even small changes to API behavior can be breaking for them, and will be considered breaking for lol-html. Lol-html has a commitment to stability; breaking changes are only to be considered as a last resort.

- [ ] My changes **do not** introduce any breaking changes or change existing API behaviors.
- [ ] My changes **do** introduce breaking changes or changes existing API behaviors.
```

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+### Description
+
+Please provide a brief description of the changes in this pull request.
+
+### Motivation
+
+Explain the motivation behind these changes and why they are necessary.
```
Comment (inikulin) [inline .github/pull_request_template.md:7]: I think we can replace description and motivation section with just asking people to create an issue first and provide those there. So, it basically will be: ensure that there's an issue with this information

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+### Type of Change
+
+- [ ] Bug fix (non-breaking change that resolves an issue)
+- [ ] Feature enhancement (non-breaking change that adds functionality)
+- [ ] Documentation update (non-breaking change that improves documentation)
+- [ ] Refactoring (non-breaking change that improves code structure)
+- [ ] Other (please specify)
```
Comment (inikulin) [inline .github/pull_request_template.md:15]: Is this useful?

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+### Checklist
+
+Please review the following checklist and ensure that all items are completed before submitting the pull request:
+
+- [ ] My changes do not introduce any breaking changes or change existing API behaviors.
+    - Keep in mind, There are rather large users of lol-html so even small changes to API behavior can be breaking
```
Comment (inikulin) [inline .github/pull_request_template.md:22]: > rather large users

this sounds quite arrogant, let's just replace with something like "big userbase"

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      for them, and will be considered breaking for lol-html. Lol-html has a commitment to stability; breaking
+      changes are only to be considered as a last resort.
+- [ ] I have tested my changes thoroughly and they do not cause any regressions.
+- [ ] My code follows the project's coding style guidelines.
```
Comment (inikulin) [inline .github/pull_request_template.md:26]: Would be nice to provide a link to those guidelines (though, I don't remember if we have them lol)

**Review by inikulin — CHANGES_REQUESTED** [review]

**Comment by Warfields** [comment]
Re @inikulin, I've gone with a much simpler version as well as implementing your feedback.

File: .github/pull_request_template.md (hunk 1 above)
Comment (Warfields) [reply .github/pull_request_template.md]: Done.

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+### Stability
+Keep in mind, there is a big userbase for lol-html so even small changes to API behavior can be
+breaking for some of them, and will be considered breaking for lol-html.
+
+**Lol-html has a commitment to stability; breaking changes are only to be considered as a last resort.**
+
+- [ ] My changes **do not** introduce any breaking changes or change existing API behaviors.
+- [ ] My changes **do** introduce breaking changes or changes existing API behaviors.
+  - [ ] I've discussed these changes with the repository maintainers and received a green light.
```
Comment (inikulin) [inline .github/pull_request_template.md:13]: I think we can just have one tick with:
> My changes **do** introduce breaking changes or changes existing API behaviors. I've discussed these changes with the repository maintainers and received a green light.

And make people tick it if that's the case.

Also, let's revive ticks for testing and documentation - those were useful. Another tick that would be useful is a reminder to update C API if changes to Rust API were made

**Review by inikulin — CHANGES_REQUESTED** [review]

**Pull request closed without merging**